src/components/NihLinkWarning.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react'
+import { div } from 'react-hyperscript-helpers'
+import { clearNotification, notify } from 'src/components/Notifications'
+import { authStore } from 'src/libs/auth'
+import * as Utils from 'src/libs/utils'
+import * as Profile from 'src/pages/Profile'
+
+
+export const NihLinkWarning = () => {
+  const { nihStatus } = Utils.useAtom(authStore)
+
+  useEffect(() => {
+    const notificationId = 'nih-link-warning'
+
+    if (nihStatus) {
+      const { linkedNihUsername, linkExpireTime } = nihStatus
+      const expireTime = linkExpireTime * 1000
+      const expired = expireTime < Date.now()
+      const expiringSoon = !expired && expireTime < (Date.now() + (1000 * 60 * 60 * 24))
+
+      if (!!linkedNihUsername && (expired || expiringSoon)) {
+        const expirationMessage = expired ? 'has expired' : 'will expire soon'
+        notify('info', div({}, [
+          `Your access to NIH Controlled Access workspaces and data ${expirationMessage} and your access to NIH Controlled Access workspaces will be revoked. `,
+          Profile.renderShibbolethLink('Re-link'),
+          ` your Terra and eRA Commons / NIH accounts (${linkedNihUsername}) to retain access to these workspaces and data.`
+        ]), { id: notificationId })
+      }
+    } else {
+      clearNotification(notificationId)
+    }
+  }, [nihStatus])
+
+  return null
```

Comment:
TOL: This pattern (effect, and render null) is good for cases where you want "pluggable" behaviors, e.g. the effect should only be active when the component is mounted. In this case, since we're rendering this globally anyway, this could also be formulated as another direct subscription to `authStore`. It's totally fine as written, though.

Reply:
I'd honestly prefer it as a direct subscription, since that would be a minor rewrite, so that it was with the other things like it.